libs/execution/src/lib/types/io-types/filesystem-node-directory.ts

```diff
@@ -0,0 +1,52 @@
+// SPDX-FileCopyrightText: 2023 Friedrich-Alexander-Universitat Erlangen-Nurnberg
+//
+// SPDX-License-Identifier: AGPL-3.0-only
+import { FileSystemNode } from './filesystem-node';
+import { FileSystemFile } from './filesystem-node-file';
+
+export class FileSystemDirectory extends FileSystemNode {
+  private children: FileSystemNode[] = [];
+  constructor(public override name: string) {
+    super(name);
+  }
+
+  override getNodeSize(): number {
+    return this.children.reduce(
+      (totalSize, child) => totalSize + child.getNodeSize(),
+      0,
+    );
+  }
+
+  add(fileSystemNode: FileSystemNode) {
+    this.children.push(fileSystemNode);
```

Comment:
You need to pay attention to duplicate names here. E.g. if a user tries to add a file `file.txt` but a file with the same name already exists as a child of this directory.

Reply:
added check if node name exists. This implementation requires now in each node to have unique node names.

Drawback is, that we cant have a dir and a file with same name in same directory, but this can be implemented later on if needed.

Reply:
I think this should work right away, so the in-memory file system is semantically equivalent to the one of the corresponding ZIP archive. In case we postpone it, we should definitely track it via an issue.

Reply:
Now it works equally, so no track in new issue neccesary.